eZ/Publish/Core/Repository/URLAliasService.php

```diff
@@ -636,21 +641,21 @@ protected function buildSPIUrlAlias(URLAlias $urlAlias)
     /**
      * looks up the URLAlias for the given url.
      *
-     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException if the path does not exist or is not valid for the given language
-     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException if the path exceeded maximum depth level
-     *
      * @param string $url
      * @param string $languageCode
+     * @param bool|null $showAllTranslations
      *
      * @return \eZ\Publish\API\Repository\Values\Content\URLAlias
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException if the path does not exist or is not valid for the given language
+     * @throws \eZ\Publish\Core\Base\Exceptions\NotFoundException
      */
-    public function lookup($url, $languageCode = null)
+    public function lookup($url, $languageCode = null, bool $showAllTranslations = null)
```

Comment:
I think `$showAllTranslations ` parameter should be also added to the `\eZ\Publish\API\Repository\URLAliasService::lookup` implementation in the SiteAccess aware layer. 

Are you able to create a test case for proposed changes?

Reply:
Working at it right now.